simulator/risc_v/t/unit_test.cpp

```diff
@@ -547,6 +547,46 @@ TEST_CASE("RISCV RV64 ror")
     }
 }
 
+TEST_CASE("RISCV RV32 rori disassembly")
+{
+    CHECK( RISCVInstr<uint32>( 0x6007D793).get_disasm() == "rori $a5, $a5, 0");  // 01100 | 0000000 (0)  | 01111 ($a5) | 101 | 01111 ($a5) | 0010011
+    CHECK( RISCVInstr<uint32>( 0x60F7D793).get_disasm() == "rori $a5, $a5, 15"); // 01100 | 0001111 (15) | 01111 ($a5) | 101 | 01111 ($a5) | 0010011
+    CHECK( RISCVInstr<uint32>( 0x61F7D793).get_disasm() == "rori $a5, $a5, 31"); // 01100 | 0011111 (31) | 01111 ($a5) | 101 | 01111 ($a5) | 0010011
+}
+
+TEST_CASE("RISCV RV64 rori disassembly")
+{
+    CHECK( RISCVInstr<uint64>( 0x63F7D793).get_disasm() == "rori $a5, $a5, 63"); // 01100 | 0111111 (63) | 01111 ($a5) | 101 | 01111 ($a5) | 0010011
+}
+
+TEST_CASE("RISCV RV32 rori")
+{
+    std::vector<TestData<uint32>> cases {
+        TestData<uint32>( 0x01234567, 0,  0x01234567),
+        TestData<uint32>( 0x01234567, 4,  0x70123456),
+        TestData<uint32>( 0x01234567, 8,  0x67012345),
+        TestData<uint32>( 0x01234567, 31, 0x02468ace),
```

Comment:
Can we test that output is valid if immediate == 32?

Reply:
Yes, we can and it works correctly, however according to documentation: "Bit addresses are 5 bit values on RV32 and 6 bit values on RV64", also disassembling decodes only 5 lower immediate bits for RV32 and 6 lower immediate bits for RV64 (notwithstanding immediate type '7' used), checked experimentally. So I decided not to add tests for values that do no fit in 5bit for RV32 and 6bit for RV64.

Reply:
I suggest adding that, because it is still a valid input — there should be no trap, no overflow, etc.
For SLLI from I-subset, the behavior is defined explicitly:

> Shifts by a constant are encoded as a specialization of the I-type format. The operand to be shifted
is in rs1, and the shift amount is encoded in the lower 5 bits of the I-immediate field

https://github.com/riscv/riscv-isa-manual/releases/download/Ratified-IMAFDQC/riscv-spec-20191213.pdf

I think it is missed in documentation for SLOI, RORI, and similar instructions. Let's open a pull request to fix it.

Reply:
Well, it has the following piece:

> These instructions are similar to shift-logical operations from the base spec, except they shift in
the values from the opposite side of the register, in order.

https://raw.githubusercontent.com/riscv/riscv-bitmanip/master/bitmanip-0.92.pdf, page 14